draft-ietf-jose-hpke-encrypt.md

```diff
@@ -345,11 +429,18 @@ HPKE also offers modes that offer authentication.
 HPKE relies on a source of randomness to be available on the device.
 In Key Agreement with Key Wrapping mode, CEK has to be randomly generated and it MUST be ensured that the guidelines in {{RFC8937}} for random number generations are followed.
 
+## HPKE authentication
+
+Authenticated HPKE modes MUST NOT be used with Key Encryption, as this is trivially insecure.
```

Comment:
Anyone that has access to CEK can alter the message by just encrypting a new one and replacing the protected headers / ciphertext.

Reply:
It is not clear to me why the Authenticated HPKE mode cannot be used 
with Key Encryption ?

Reply:
Put another way, using Authenticated HPKE with Key Encryption would authenticate the CEK. Which does not even indirectly authenticate the actual message.

Whereas Authenticated HPKE with Integrated Encryption does authenticate protected headers, JWE AAD and the  message plaintext.

Reply:
If there are multiple recipients in the original message, the message authentication breaks down since any other recipient might have forged the message.

Checking for one recipient does not work: The number of recipients is not authenticated, so attacker might have stripped all recipients except the target, so the message appears to have just a single recipient.

Reply:
Yes, if any recipient acts as a MiTM to modify the message or strip the recipient list, such an attack cannot be detected in JWE, even when Authenticated HPKE is not used.

Reply:
Such message does not even pretend to be authenticated (unless done by external means like signatures).

Reply:
I propose to modify the text as follows:
Authenticated HPKE modes MUST NOT be used for Key Encryption, as the message is not authenticated. Any recipient could act as a man-in-the-middle (MitM) and modify the message.